request_handler.cpp

```diff
@@ -73,7 +73,7 @@ void request_handler::handle_request(const char buffer_[8192], reply& rep)
 
   // Fill out the reply to be sent to the client.
   rep.status = reply::ok;
-  for (int i =0; buffer_[i] != '\0' || i < 8192; i++)
+  for (int i =0; buffer_[i] != '\0' && i < 8192; i++)
```

Comment:
We should probably check `i < 8192` before evaluating `buffer_[i] != '\0'`, because otherwise you run the risk of accessing memory out of bounds